### ISSUE TITLE
Move steps requiring root password at the beginning

### ISSUE THREAD

**Issue by Crocmagnon** [topgrade-rs/topgrade#376]
## I want to suggest some general feature
Topgrade should run the steps requiring passwords at the beginning, so that users can leave an upgrade mostly unattended after it has started.

For example on macOS : `tlmgr` requires the root password and in my system it is typically run in the middle of the process:

```
── 08:47:15 - Summary ──────────────────────────────────────────────────────────
Brew (ARM): OK
Brew Cask (ARM): OK
fisher: OK
TLDR: OK
rustup: OK
cargo: OK
pipx: OK
tlmgr: OK
chezmoi: OK
npm: OK
Containers: OK
krew: OK
GitHub CLI Extensions: OK
System upgrade: OK
```

## More information
Test process:
* Run topgrade on a system where tlmgr is available and enabled
* tlmgr should run at the beginning

Same goes for all steps requiring a password.

**Comment by Crocmagnon** [comment]
Yes, I guess that would also solve the issue in a nice and easier way 👍🏻

**Comment by GideonBear** [comment]
Closing in favor of using `pre_sudo = true`, and #1049 for the priorities.
